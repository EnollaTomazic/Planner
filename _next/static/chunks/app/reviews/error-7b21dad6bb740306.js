(self.webpackChunk_N_E=self.webpackChunk_N_E||[]).push([[6741],{13797:(e,r,t)=>{"use strict";t.r(r),t.d(r,{default:()=>a});var n=t(95155),i=t(69975);let a=function(e){return(0,n.jsx)(i.R,{...e,title:"Reviews hit a bump",description:"We couldn't open your review vault. Give it another try to resume editing."})}},41597:(e,r,t)=>{Promise.resolve().then(t.bind(t,13797))},69975:(e,r,t)=>{"use strict";t.d(r,{R:()=>A,A:()=>k});var n,i,a,l,s,o,u,c=t(95155),d=t(12115),p=t(52619),v=t.n(p),h=t(91182),m=t(10161),N=t(87358);let f=m.UG.child("sentry"),E=f.child("capture"),x=(null==(n=N.env.NEXT_PUBLIC_SENTRY_DSN)?void 0:n.trim())||(null==(i=N.env.SENTRY_DSN)?void 0:i.trim())||"",y=(null==(a=N.env.NEXT_PUBLIC_SENTRY_ENVIRONMENT)?void 0:a.trim())||(null==(l=N.env.SENTRY_ENVIRONMENT)?void 0:l.trim())||"production",S=(null==(s=N.env.NEXT_PUBLIC_SENTRY_TRACES_SAMPLE_RATE)?void 0:s.trim())||(null==(o=N.env.SENTRY_TRACES_SAMPLE_RATE)?void 0:o.trim())||"",_=!!x,R=!1,T=null,b=null,g=null!=(u=function(e){if(!e)return;let r=Number.parseFloat(e);return Number.isNaN(r)?void f.warn("Invalid Sentry sample rate provided",{value:e}):Math.min(Math.max(r,0),1)}(S))?u:.1;async function w(){return _?b||(T||(T=Promise.all([t.e(4850),t.e(7141)]).then(t.bind(t,17141)).then(e=>{if(b=e,!R)try{e.init({dsn:x,environment:y,enabled:_,tracesSampleRate:g,replaysSessionSampleRate:0,replaysOnErrorSampleRate:0,profilesSampleRate:0}),R=!0}catch(e){return f.error("Failed to initialize Sentry",e),null}return b}).catch(e=>(f.error("Failed to load Sentry runtime",e),b=null,null)).finally(()=>{T=null})),T):null}async function j(e,r){if(!_)return;let t=await w();if(!t)return void f.warn("captureException skipped because Sentry is unavailable",e);try{t.captureException(e,r)}catch(r){E.error("Failed to capture exception with Sentry",r,e)}}var C=t(27566);let P=(0,m.h)("route:error-boundary");function A(e){let{error:r,reset:t,title:n,description:i,retryLabel:a="Try again",homeLabel:l="Go to dashboard",homeHref:s=(0,C.hd)("/")}=e;return(0,d.useEffect)(()=>{P.error("Route error boundary captured an exception",r),j(r,{tags:{boundary:"route"},extra:{digest:null==r?void 0:r.digest}})},[r]),(0,c.jsx)(h.qK,{as:"section",role:"alert","aria-live":"assertive",className:"py-[var(--space-8)]",children:(0,c.jsxs)("div",{className:"flex max-w-prose flex-col gap-[var(--space-4)]",children:[(0,c.jsxs)("div",{className:"space-y-[var(--space-2)]",children:[(0,c.jsx)("h1",{className:"text-title-lg font-semibold text-foreground",children:n}),(0,c.jsx)("p",{className:"text-body text-muted-foreground",children:i})]}),(0,c.jsxs)("div",{className:"flex flex-wrap gap-[var(--space-3)]",children:[(0,c.jsx)(h.$n,{variant:"default",onClick:t,children:a}),(0,c.jsx)(h.$n,{asChild:!0,variant:"quiet",children:(0,c.jsx)(v(),{href:s,children:l})})]})]})})}function k(e){return(0,c.jsx)(A,{...e,title:"Something went wrong",description:"This section hit an error, but the rest of Planner is still running. Try again or return home."})}}},e=>{e.O(0,[2790,5830,1685,7766,2619,1182,8441,1255,7358],()=>e(e.s=41597)),_N_E=e.O()}]);