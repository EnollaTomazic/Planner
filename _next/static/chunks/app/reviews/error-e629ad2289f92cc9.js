(self.webpackChunk_N_E=self.webpackChunk_N_E||[]).push([[6741],{13797:(e,r,s)=>{"use strict";s.r(r),s.d(r,{default:()=>i});var t=s(95155),a=s(50960);let i=function(e){return(0,t.jsx)(a.R,{...e,title:"Reviews hit a bump",description:"We couldn't open your review vault. Give it another try to resume editing."})}},41597:(e,r,s)=>{Promise.resolve().then(s.bind(s,13797))},50960:(e,r,s)=>{"use strict";s.d(r,{A:()=>c,R:()=>o});var t=s(95155),a=s(12115),i=s(52619),n=s.n(i),l=s(24625);function o(e){let{error:r,reset:s,title:i,description:o,retryLabel:c="Try again",homeLabel:d="Go to dashboard",homeHref:h="/"}=e;return(0,a.useEffect)(()=>{console.error(r)},[r]),(0,t.jsx)(l.qK,{as:"section",role:"alert","aria-live":"assertive",className:"py-[var(--space-8)]",children:(0,t.jsxs)("div",{className:"flex max-w-prose flex-col gap-[var(--space-4)]",children:[(0,t.jsxs)("div",{className:"space-y-[var(--space-2)]",children:[(0,t.jsx)("h1",{className:"text-display-sm font-semibold text-foreground",children:i}),(0,t.jsx)("p",{className:"text-body text-muted-foreground",children:o})]}),(0,t.jsxs)("div",{className:"flex flex-wrap gap-[var(--space-3)]",children:[(0,t.jsx)(l.$n,{variant:"primary",onClick:s,children:c}),(0,t.jsx)(l.$n,{asChild:!0,variant:"ghost",children:(0,t.jsx)(n(),{href:h,children:d})})]})]})})}function c(e){return(0,t.jsx)(o,{...e,title:"Something went wrong",description:"This section hit an error, but the rest of Planner is still running. Try again or return home."})}}},e=>{e.O(0,[6428,9295,2619,4625,8441,1255,7358],()=>e(e.s=41597)),_N_E=e.O()}]);