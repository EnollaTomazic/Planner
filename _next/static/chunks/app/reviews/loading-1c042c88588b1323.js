(self.webpackChunk_N_E=self.webpackChunk_N_E||[]).push([[5883],{18515:(a,s,e)=>{Promise.resolve().then(e.bind(e,71812))},71812:(a,s,e)=>{"use strict";e.d(s,{default:()=>i});var c=e(95155),r=e(24625);let d=["filter-sort","filter-tags"],l=["review-1","review-2","review-3","review-4","review-5"],p=["summary","edit"];function i(){return(0,c.jsxs)(c.Fragment,{children:[(0,c.jsx)("header",{className:"page-shell py-[var(--space-6)]",children:(0,c.jsxs)("div",{className:"rounded-card card-neo-soft shadow-neo-strong p-[var(--space-4)] space-y-[var(--space-4)]",children:[(0,c.jsxs)("div",{className:"space-y-[var(--space-2)]",children:[(0,c.jsx)(r.EA,{className:"h-[var(--space-3)] w-1/4",radius:"sm"}),(0,c.jsx)(r.EA,{className:"h-[var(--space-5)] w-1/3",radius:"sm"})]}),(0,c.jsxs)("div",{className:"grid gap-[var(--space-3)] md:grid-cols-12",children:[(0,c.jsx)(r.EA,{className:"md:col-span-8 h-[calc(var(--space-8)+var(--space-2))]",radius:"card"}),(0,c.jsxs)("div",{className:"md:col-span-2 space-y-[var(--space-2)]",children:[(0,c.jsx)(r.EA,{className:"h-[var(--space-3)] w-1/2",radius:"sm"}),(0,c.jsx)(r.EA,{className:"h-[var(--space-8)]",radius:"card"})]}),(0,c.jsx)(r.EA,{className:"md:col-span-2 h-[var(--space-8)]",radius:"card"})]})]})}),(0,c.jsx)("main",{className:"page-shell space-y-[var(--space-6)] py-[var(--space-6)]","aria-busy":"true",children:(0,c.jsxs)("section",{className:"grid gap-[var(--space-4)] md:grid-cols-6 lg:grid-cols-12",children:[(0,c.jsxs)("aside",{className:"space-y-[var(--space-3)] md:col-span-2 lg:col-span-4",children:[(0,c.jsxs)("div",{className:"rounded-card card-neo-soft shadow-neo-strong p-[var(--space-3)] space-y-[var(--space-3)]",children:[(0,c.jsx)(r.EA,{className:"h-[var(--space-3)] w-2/3",radius:"sm"}),(0,c.jsx)("div",{className:"space-y-[var(--space-2)]",children:d.map(a=>(0,c.jsx)(r.EA,{className:"h-[var(--space-6)]",radius:"card"},a))})]}),(0,c.jsx)("div",{className:"rounded-card card-neo-soft shadow-neo-strong p-[var(--space-3)] space-y-[var(--space-2)]",children:l.map(a=>(0,c.jsxs)("div",{className:"rounded-card border border-border/40 p-[var(--space-3)] space-y-[var(--space-2)]",children:[(0,c.jsx)(r.EA,{className:"h-[var(--space-4)] w-2/3",radius:"sm"}),(0,c.jsx)(r.EA,{className:"w-5/6"})]},a))})]}),(0,c.jsxs)("section",{className:"space-y-[var(--space-4)] md:col-span-4 lg:col-span-8",children:[(0,c.jsx)("div",{className:"flex flex-wrap gap-[var(--space-2)]",children:p.map(a=>(0,c.jsx)(r.EA,{className:"h-[var(--space-6)] w-[calc(var(--space-8)*2)]",radius:"full"},a))}),(0,c.jsxs)("article",{className:"rounded-card card-neo-soft shadow-neo-strong p-[var(--space-4)] space-y-[var(--space-3)]",children:[(0,c.jsx)(r.EA,{className:"h-[var(--space-5)] w-1/2",radius:"sm"}),(0,c.jsx)(r.EA,{className:"w-full"}),(0,c.jsx)(r.EA,{className:"w-5/6"}),(0,c.jsxs)("div",{className:"grid gap-[var(--space-3)] sm:grid-cols-2",children:[(0,c.jsx)(r.EA,{className:"h-[calc(var(--space-8)+var(--space-4))]",radius:"card"}),(0,c.jsx)(r.EA,{className:"h-[calc(var(--space-8)+var(--space-4))]",radius:"card"})]})]})]})]})})]})}}},a=>{a.O(0,[6428,9295,4625,8441,1255,7358],()=>a(a.s=18515)),_N_E=a.O()}]);