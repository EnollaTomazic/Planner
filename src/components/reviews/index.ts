// src/components/reviews/index.ts
export { default as ReviewPage } from "./ReviewPage";
export { default as ReviewsPage } from "./ReviewsPage";
export { default as ReviewEditor } from "./ReviewEditor";
export { default as ReviewList } from "./ReviewList";
export { default as ReviewListItem } from "./ReviewListItem";
export { default as ReviewSummary } from "./ReviewSummary";
export { default as ReviewSummaryHeader } from "./ReviewSummaryHeader";
export { default as ReviewSummaryScore } from "./ReviewSummaryScore";
export { default as ScoreMeter } from "./ScoreMeter";
export { default as ReviewSummaryPillars } from "./ReviewSummaryPillars";
export { default as ReviewSummaryTimestamps } from "./ReviewSummaryTimestamps";
export { default as ReviewSummaryNotes } from "./ReviewSummaryNotes";
export { default as RoleSelector } from "./RoleSelector";
export { default as ReviewPanel } from "./ReviewPanel";
export { default as NeonIcon } from "./NeonIcon";
export { useReviewFilter } from "./useReviewFilter";
export { default as LaneOpponentForm } from "./LaneOpponentForm";
export { default as ResultScoreSection } from "./ResultScoreSection";
export { default as PillarsSelector } from "./PillarsSelector";
export { default as TimestampMarkers } from "./TimestampMarkers";
