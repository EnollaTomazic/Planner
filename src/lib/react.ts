import * as React from "react";

export const hasTextContent = (node: React.ReactNode): boolean => {
  if (node === null || node === undefined) return false;
  if (typeof node === "boolean") return false;
  if (typeof node === "string") return node.trim().length > 0;
  if (typeof node === "number") return true;
  if (Array.isArray(node)) return node.some((item) => hasTextContent(item));
  if (React.isValidElement<{ children?: React.ReactNode }>(node)) {
    return hasTextContent(node.props.children ?? null);
  }
  return false;
};

export function observeMediaQuery(
  query: string,
  onChange: (matches: boolean, mediaQuery: MediaQueryList) => void,
): (() => void) | undefined {
  if (typeof window === "undefined" || typeof window.matchMedia !== "function") {
    return undefined;
  }

  let mediaQuery: MediaQueryList;

  try {
    mediaQuery = window.matchMedia(query);
  } catch {
    return undefined;
  }

  const handleChange = (event: MediaQueryListEvent): void => {
    onChange(event.matches, mediaQuery);
  };

  const handleLegacyChange = (): void => {
    onChange(mediaQuery.matches, mediaQuery);
  };

  onChange(mediaQuery.matches, mediaQuery);

  if (typeof mediaQuery.addEventListener === "function") {
    mediaQuery.addEventListener("change", handleChange);
    return () => {
      mediaQuery.removeEventListener("change", handleChange);
    };
  }

  if (typeof mediaQuery.addListener === "function") {
    mediaQuery.addListener(handleLegacyChange);
    return () => {
      mediaQuery.removeListener(handleLegacyChange);
    };
  }

  return undefined;
}

export function useMatchMedia(
  query: string,
  getServerSnapshot?: () => boolean,
): boolean {
  const getMatches = React.useCallback(() => {
    if (typeof window === "undefined") {
      return typeof getServerSnapshot === "function" ? getServerSnapshot() : false;
    }

    if (typeof window.matchMedia !== "function") {
      return false;
    }

    try {
      return window.matchMedia(query).matches;
    } catch {
      return false;
    }
  }, [query, getServerSnapshot]);

  const [matches, setMatches] = React.useState<boolean>(getMatches);

  React.useEffect(() => {
    return observeMediaQuery(query, (nextMatches) => {
      setMatches(nextMatches);
    });
  }, [query]);

  return matches;
}
