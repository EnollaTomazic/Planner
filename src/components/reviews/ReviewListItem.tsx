// src/components/reviews/ReviewListItem.tsx
"use client";

import * as React from "react";
import { cn } from "@/lib/utils";
import type { Review } from "@/lib/types";
import { Badge } from "@/components/ui";

const shellBase = cn(
  "relative w-full text-left rounded-card r-card-lg p-3 bg-[hsl(var(--card)/0.9)] border border-[hsl(var(--border)/0.35)] transition-all duration-200 focus-visible:outline-none disabled:opacity-60 disabled:pointer-events-none",
  "hover:ring-2 hover:ring-[--theme-ring]",
  "data-[selected=true]:ring-2 data-[selected=true]:ring-[--theme-accent]",
);

const statusDotBase = "h-2 w-2 rounded-full ring-2";
const statusDotWin = "bg-success ring-success";
const statusDotLoss = "bg-danger ring-danger";
const statusDotDefault = "bg-muted-foreground ring-muted-foreground";
const statusDotPulse =
  "motion-safe:animate-[pulse_2s_ease-in-out_infinite] motion-reduce:animate-none";
const statusDotBlink =
  "motion-safe:animate-[blink_1s_steps(2)_infinite] motion-reduce:animate-none";
const itemLoading = cn(
  shellBase,
  "motion-safe:animate-pulse motion-reduce:animate-none",
);
const loadingLine = "h-3 rounded-md bg-muted";
const scoreBadge =
  "px-2 py-0.5 rounded-full text-xs leading-none font-medium text-background bg-gradient-to-br from-[--theme-accent] to-[--theme-accent2] ring-1 ring-[--theme-accent] [--ring:var(--theme-accent)]";

export type ReviewListItemProps = {
  review?: Review;
  selected?: boolean;
  disabled?: boolean;
  loading?: boolean;
  onClick?: () => void;
};

export default function ReviewListItem({
  review,
  selected = false,
  disabled = false,
  loading = false,
  onClick,
}: ReviewListItemProps) {
  if (loading) {
    return (
      <div data-scope="reviews" className={itemLoading}>
        <div className={cn(loadingLine, "w-3/5 mb-3")} />
        <div className={cn(loadingLine, "w-2/5")} />
      </div>
    );
  }

  const title = review?.title?.trim() || "Untitled Review";
  const untitled = !review?.title?.trim();
  const matchup = review?.matchup?.trim() || "";
  const score = review?.score;
  const role = review?.role;

  return (
    <button
      data-scope="reviews"
      type="button"
      disabled={disabled}
      onClick={onClick}
      aria-label={`Open review: ${title}`}
      data-selected={selected ? "true" : undefined}
      className={shellBase}
    >
      <div className="flex flex-col gap-2">
        <div className="flex items-start justify-between gap-2">
          <div
            className={cn(
              "truncate font-medium text-base",
              untitled && "text-muted-foreground/70",
            )}
            aria-label={untitled ? "Untitled Review" : undefined}
          >
            {title}
          </div>
          {matchup ? (
            <div className="truncate text-sm text-muted-foreground">
              {matchup}
            </div>
          ) : null}
        </div>
        <div className="flex items-center justify-between gap-2">
          <div className="flex items-center gap-2">
            <span
              aria-hidden
              className={cn(
                statusDotBase,
                statusDotBlink,
                review?.result === "Win"
                  ? statusDotWin
                  : review?.result === "Loss"
                    ? statusDotLoss
                    : statusDotDefault,
                review?.status === "new" && statusDotPulse,
              )}
            />
            {role ? (
              <Badge
                variant="neutral"
                className="px-1 py-0 text-xs leading-none"
              >
                {role}
              </Badge>
            ) : null}
          </div>
          {typeof score === "number" ? (
            <span
              className={scoreBadge}
              aria-label={`Rating ${score} out of 10`}
            >
              {score}/10
            </span>
          ) : null}
        </div>
      </div>
    </button>
  );
}
