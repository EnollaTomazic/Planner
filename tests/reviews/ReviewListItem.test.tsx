import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { describe, it, expect, afterEach } from 'vitest';
import { ReviewListItem } from '../../src/components/reviews/ReviewListItem';
import type { Review } from '../../src/lib/types';

afterEach(cleanup);

const baseReview: Review = {
  id: '1',
  title: 'Sample Review',
  notes: 'Quick note',
  createdAt: 1700000000000,
  matchup: 'Lux vs Ahri',
  role: 'MID',
  score: 9,
  result: 'Win',
  tags: [],
  pillars: [],
};

describe('ReviewListItem', () => {
  it('renders default state', () => {
    const { container } = render(<ReviewListItem review={baseReview} />);
    expect(container).toMatchSnapshot();
  });

  it('renders selected state', () => {
    const { container } = render(
      <ReviewListItem review={baseReview} selected />
    );
    expect(container).toMatchSnapshot();
  });

  it('renders loading state', () => {
    const { container } = render(<ReviewListItem loading />);
    expect(container).toMatchSnapshot();
  });

  it('renders disabled state', () => {
    const { container } = render(
      <ReviewListItem disabled review={baseReview} />,
    );
    expect(container.firstChild).toHaveClass(
      'disabled:opacity-disabled',
      'disabled:pointer-events-none',
    );
    expect(container).toMatchSnapshot();
  });

  it('renders untitled state', () => {
    const review = { ...baseReview, title: '' };
    const { container } = render(<ReviewListItem review={review} />);
    expect(container).toMatchSnapshot();
  });
});
