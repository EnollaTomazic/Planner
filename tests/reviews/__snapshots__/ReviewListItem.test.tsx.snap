// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ReviewListItem > renders default state 1`] = `
<div>
  <button
    aria-label="Open review: Sample Review"
    class="_tile_d828d5 relative w-full text-left h-auto min-h-[76px] p-4 rounded-lg border border-border/50 bg-card/60 scanlines noise jitter transition-all duration-200 hover:-translate-y-[1px] focus-visible:outline-none disabled:opacity-60 disabled:pointer-events-none"
    data-scope="reviews"
    type="button"
  >
    <div
      class="grid grid-cols-[20px_1fr_auto] gap-3"
    >
      <span
        aria-hidden="true"
        class="self-center justify-self-center h-2 w-2 rounded-full motion-safe:animate-[blink_1s_steps(2)_infinite] motion-reduce:animate-none bg-success"
      />
      <div
        class="min-w-0 flex flex-col gap-1"
      >
        <div
          class="truncate font-medium text-base"
        >
          Sample Review
        </div>
        <div
          class="flex items-center gap-2 text-sm text-muted-foreground"
        >
          <span
            aria-label="Rating 9 out of 10"
            class="inline-flex items-center text-xs font-medium border rounded-md px-2 py-1 bg-accent/15 border-accent/35 text-accent shadow-[0_0_8px_hsl(var(--accent)/0.3)]"
          >
            9
            /10
          </span>
          <span
            class="line-clamp-1 truncate"
          >
            Quick note
          </span>
        </div>
      </div>
      <span
        class="self-start text-xs text-muted-foreground whitespace-nowrap"
      >
        11/14/2023
      </span>
    </div>
  </button>
</div>
`;

exports[`ReviewListItem > renders loading state 1`] = `
<div>
  <div
    class="_tile_d828d5 relative w-full text-left h-auto min-h-[76px] p-4 rounded-lg border border-border/50 bg-card/60 scanlines noise jitter transition-all duration-200 hover:-translate-y-[1px] focus-visible:outline-none disabled:opacity-60 disabled:pointer-events-none motion-safe:animate-pulse motion-reduce:animate-none"
    data-scope="reviews"
  >
    <div
      class="h-3 rounded bg-muted w-3/5 mb-3"
    />
    <div
      class="h-3 rounded bg-muted w-2/5"
    />
  </div>
</div>
`;

exports[`ReviewListItem > renders selected state 1`] = `
<div>
  <button
    aria-label="Open review: Sample Review"
    class="_tile_d828d5 relative w-full text-left h-auto min-h-[76px] p-4 rounded-lg border border-border/50 bg-card/60 scanlines noise jitter transition-all duration-200 hover:-translate-y-[1px] focus-visible:outline-none disabled:opacity-60 disabled:pointer-events-none _tileActive_d828d5"
    data-scope="reviews"
    data-selected="true"
    type="button"
  >
    <div
      class="grid grid-cols-[20px_1fr_auto] gap-3"
    >
      <span
        aria-hidden="true"
        class="self-center justify-self-center h-2 w-2 rounded-full motion-safe:animate-[blink_1s_steps(2)_infinite] motion-reduce:animate-none bg-success"
      />
      <div
        class="min-w-0 flex flex-col gap-1"
      >
        <div
          class="truncate font-medium text-base"
        >
          Sample Review
        </div>
        <div
          class="flex items-center gap-2 text-sm text-muted-foreground"
        >
          <span
            aria-label="Rating 9 out of 10"
            class="inline-flex items-center text-xs font-medium border rounded-md px-2 py-1 bg-accent/15 border-accent/35 text-accent shadow-[0_0_8px_hsl(var(--accent)/0.3)]"
          >
            9
            /10
          </span>
          <span
            class="line-clamp-1 truncate"
          >
            Quick note
          </span>
        </div>
      </div>
      <span
        class="self-start text-xs text-muted-foreground whitespace-nowrap"
      >
        11/14/2023
      </span>
    </div>
  </button>
</div>
`;

exports[`ReviewListItem > renders untitled state 1`] = `
<div>
  <button
    aria-label="Open review: Untitled Review"
    class="_tile_d828d5 relative w-full text-left h-auto min-h-[76px] p-4 rounded-lg border border-border/50 bg-card/60 scanlines noise jitter transition-all duration-200 hover:-translate-y-[1px] focus-visible:outline-none disabled:opacity-60 disabled:pointer-events-none"
    data-scope="reviews"
    type="button"
  >
    <div
      class="grid grid-cols-[20px_1fr_auto] gap-3"
    >
      <span
        aria-hidden="true"
        class="self-center justify-self-center h-2 w-2 rounded-full motion-safe:animate-[blink_1s_steps(2)_infinite] motion-reduce:animate-none bg-success"
      />
      <div
        class="min-w-0 flex flex-col gap-1"
      >
        <div
          aria-label="Untitled Review"
          class="truncate font-medium text-base text-muted-foreground/70"
        >
          Untitled Review
        </div>
        <div
          class="flex items-center gap-2 text-sm text-muted-foreground"
        >
          <span
            aria-label="Rating 9 out of 10"
            class="inline-flex items-center text-xs font-medium border rounded-md px-2 py-1 bg-accent/15 border-accent/35 text-accent shadow-[0_0_8px_hsl(var(--accent)/0.3)]"
          >
            9
            /10
          </span>
          <span
            class="line-clamp-1 truncate"
          >
            Quick note
          </span>
        </div>
      </div>
      <span
        class="self-start text-xs text-muted-foreground whitespace-nowrap"
      >
        11/14/2023
      </span>
    </div>
  </button>
</div>
`;
